config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigStore.java

```diff
@@ -42,6 +43,17 @@ UpsertedConfig writeConfig(
    */
   ContextSpecificConfig getConfig(ConfigResourceContext configResourceContext) throws IOException;
 
+  /**
+   * Get all the configs with the latest version(along with the context to which it applies) for the
+   * specified resource keys
+   *
+   * @param configResourceContexts
+   * @return the configs
+   * @throws IOException
+   */
+  Map<ConfigResourceContext, ContextSpecificConfig> getAllContextConfigs(
```

Comment:
```suggestion
  Map<ConfigResourceContext, ContextSpecificConfig> getContextConfigs(
```

It's a plural get, but you're filtering so it's not a "get all".